packages/main/test-integration/node/cache-content.test.ts

```diff
@@ -0,0 +1,266 @@
+      originalTtlSeconds * 1000,
+    );
+    expect(expireTime - updatedTime).to.be.lessThanOrEqual(
+      newTtlSeconds * 1000,
+    );
+  });
+  it("cacheManager update expireTime then get", async () => {
+    // cacheManager create
+    const cacheManager = new GoogleAICacheManager(
+      process.env.GEMINI_API_KEY || "",
+      endpointConfiguration,
+    );
+    const createCacheResult = await cacheManager.create({
+      ttlSeconds: 20,
+      model: "models/gemini-1.5-pro",
+      contents: [
+        {
+          role: "user",
+          parts: [{ text }],
+        },
+      ],
+    });
+    expect(createCacheResult.name).to.exist;
+    expect(createCacheResult.name.startsWith("cachedContents/")).to.be.true;
+    expect(createCacheResult.displayName).to.equal("");
+
+    // cacheManager.update
+    const newExpirationTime = new Date(new Date().getTime() + 30 * 1000);
+    const updateParams = {
+      cachedContent: { expireTime: newExpirationTime.toISOString() },
+    };
+    const updateResult = await cacheManager.update(
+      createCacheResult.name,
+      updateParams,
+    );
+
+    // cacheManager.get
+    const cache = await cacheManager.get(createCacheResult.name);
+    expect(cache.name).to.equal(createCacheResult.name);
+    expect(cache.name).to.equal(updateResult.name);
+    expect(cache.model).to.equal(createCacheResult.model);
+    expect(cache.model).to.equal(updateResult.model);
+    expect(cache.createTime).to.equal(createCacheResult.createTime);
+    expect(cache.updateTime).to.not.equal(createCacheResult.updateTime);
+    expect(cache.updateTime).to.equal(updateResult.updateTime);
+    expect(cache.expireTime).to.not.equal(createCacheResult.expireTime);
+    expect(cache.expireTime).to.equal(updateResult.expireTime);
+  });
+  it("generateContentWithCache", async () => {
+    const cacheManager = new GoogleAICacheManager(
+      process.env.GEMINI_API_KEY || "",
+      endpointConfiguration,
+    );
+    const createCacheResult = await cacheManager.create({
+      ttlSeconds: 20,
+      model: "models/gemini-1.5-pro",
+      contents: [
+        {
+          role: "user",
+          parts: [{ text }],
+        },
+      ],
+    });
+    expect(createCacheResult.name).to.exist;
+    const cacheName = createCacheResult.name;
+    expect(cacheName.startsWith("cachedContents/")).to.be.true;
+
+    // cacheManager.get
+    const cache = await cacheManager.get(cacheName);
+
+    // generate content.
+    const genAI = new GoogleGenerativeAI(process.env.GEMINI_API_KEY);
+    const model = genAI.getGenerativeModelFromCachedContent(
+      cache,
+      endpointConfiguration,
+    );
+    const result = await model.generateContent({
+      contents: [
+        {
+          role: "user",
+          parts: [{ text: "What do purple cats drink?" }],
+        },
+      ],
+    });
+    const response = await result.response;
+    expect(response.text().toLowerCase().includes("gatorade")).to.be.true;
```

Comment:
Could we configure the model to give us more predictable outputs so that this test doesn't flake as often? It looks like Andrew did something like this in the swift SDK https://github.com/google-gemini/generative-ai-swift/blob/main/Tests/GoogleAITests/GoogleAITests.swift#L26

We do tell the model the answer 1500 times so maybe it's not necessary for this test 😅

Reply:
Yeah, I haven't had any flakes in this scernario because the content cache is so specific, but I do argree that we should go back and alter some of our older tests.